pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractJavaRule.java

```diff
@@ -4,17 +4,21 @@
 
 package net.sourceforge.pmd.lang.java.rule;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import net.sourceforge.pmd.RuleContext;
 import net.sourceforge.pmd.lang.LanguageRegistry;
 import net.sourceforge.pmd.lang.ast.Node;
 import net.sourceforge.pmd.lang.java.JavaLanguageModule;
 import net.sourceforge.pmd.lang.java.ast.*;
+import net.sourceforge.pmd.lang.java.typeresolution.TypeHelper;
 import net.sourceforge.pmd.lang.rule.AbstractRule;
 import net.sourceforge.pmd.lang.rule.ImmutableLanguage;
 
-public abstract class AbstractJavaRule extends AbstractRule implements JavaParserVisitor, ImmutableLanguage {
+public abstract class AbstractJavaRule extends AbstractRule implements JavaParserVisitor, ImmutableLanguage, Annotateable {
```

Comment:
As per #956, the idea of `Annotateable` was to have the AST nodes that *are* annotateable implement it, not rules (you actually put the interface under the AST package). The interface works both as a marker interface (indicating a node *can* have annotations), and as a way to transverse the AST to get them (you here assume it's always a sibling)